### PR TITLE
Publish logs per metric and remove faulty metric filter behavior

### PR DESCRIPTION
**Pull request by silv-io**
Removes filter behavior which isn't in [documented in the official docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html). In the community edition the change in the PR amounts to filters selecting all logs all the time. Proper filtering for the JSON filters will be added to Pro. 

Loosely related to #4826

### CODE REVIEW

File: localstack/services/cloudwatch/cloudwatch_starter.py [hunk 1]
```diff
@@ -14,6 +16,22 @@ def apply_patches():
             )
         )
 
+    def list_metrics(self, *args, **kwargs):
```
Comment (thrau) [inline localstack/services/cloudwatch/cloudwatch_starter.py:19]: could we use the @patch utility here?

File: localstack/services/logs/logs_listener.py [hunk 2]
```diff
@@ -68,31 +67,28 @@ def publish_log_metrics_for_events(data):
...
-            return True
+        transformations = metric_filter.get("metricTransformations", [])
+        matches = get_pattern_matcher(pattern)
+        for log_event in log_events:
+            if matches(pattern, log_event):
+                for tf in transformations:
+                    value = tf.get("metricValue") or "1"
+                    if "$size" in value:
+                        LOG.info(
+                            "Expression not yet supported for log filter metricValue: %s" % value
```
Comment (thrau) [inline localstack/services/logs/logs_listener.py:78]: nit: i know this was here before :-) but we could use commas for varargs in logging statements, `LOG.info("...: %s", value)`

File: localstack/services/logs/logs_listener.py [hunk 3]
```diff
@@ -68,31 +67,28 @@ def publish_log_metrics_for_events(data):
...
+                        )
+                    value = float(value) if is_number(value) else 1
+                    data = [{"MetricName": tf["metricName"], "Value": value}]
+                    try:
+                        client.put_metric_data(Namespace=tf["metricNamespace"], MetricData=data)
+                    except Exception as e:
+                        LOG.info(
+                            "Unable to put metric data for matching CloudWatch log events: %s" % e
```
Comment (thrau) [inline localstack/services/logs/logs_listener.py:86]: here as well

**Review by thrau — APPROVED** [review]
great set of changes! glad we're getting this over the line

**Review by whummer — APPROVED** [review]
Nice! Great to see this getting over the line @silv-io 🚀 . Also great job on the refactoring of the tests 👍

**Pull request merged**